### DISCUSSION TITLE
Applying a circular mask to an image

### DISCUSSION THREAD

**Discussion by julien-c** [ImageMagick/ImageMagick#2169] [Help] [answered]
I would like to apply a circle mask to a PNG image. Any idea on how to use it using `convert`?

**Comment by GeeMack** [comment, marked as answer]
There are many ways to apply masks with ImageMagick. Using v6 you can make a circle cut-out from an image with a command like this. Where I used IM's built-in image "logo:", use the name of your image...

    convert logo: -alpha set -background none -fill white ^
       ( +clone -channel A -evaluate set 0 +channel -draw "circle 400,200 400,100" ) ^
       -compose dstin -composite result.png

That reads the image, makes a clone copy of it inside the parentheses, makes that clone transparent, and draws a white circle centered at 400x200. Then after the parentheses it composites the mask using a compose method of "dstin" to create the final image, transparent background with a circular piece of the input image.

If you want a hole punched out of the image instead, use the compose method "dstout".

That command is in Windows syntax. To use it in a *nix shell, change the continued-line carets "^" to backslashes "\\" and escape the parentheses "(...)" with backslashes "\\(...\\)".
